ecs-cli/modules/cli/compose/project/project.go

```diff
@@ -115,28 +115,33 @@ func (p *ecsProject) Parse() error {
 	return p.transformTaskDefinition()
 }
 
-// parseCompose loads and parses the compose yml files
+// parseCompose sets data from the compose files on the ecsProject
 func (p *ecsProject) parseCompose() error {
-	// load the compose files using libcompose
 	logrus.Debug("Parsing the compose yaml...")
+	// libcompose.Project#Parse populates project information based on its
+	// context. It sets up the name, the composefile and the composebytes
+	// (the composefile content).
 	if err := p.Project.Parse(); err != nil {
 		return err
 	}
 
 	// libcompose sanitizes the project name and removes any non alpha-numeric character.
-	// The following undo-es that and sets the project name as user defined it.
+	// The following undoes that and sets the project name as user defined it.
 	return p.context.SetProjectName()
 }
 
-// transformTaskDefinition converts the compose yml into task definition
+// transformTaskDefinition converts the compose yml and ecs-fields yml into an ECS task definition
 func (p *ecsProject) transformTaskDefinition() error {
 	context := p.context
 
 	// convert to task definition
 	logrus.Debug("Transforming yaml to task definition...")
 	taskDefinitionName := utils.GetTaskDefinitionName(context.ECSParams.ComposeProjectNamePrefix, context.Context.ProjectName)
+
 	taskRoleArn := context.CLIContext.GlobalString(command.TaskRoleArnFlag)
-	taskDefinition, err := utils.ConvertToTaskDefinition(taskDefinitionName, &context.Context, p.ServiceConfigs(), taskRoleArn)
+	ecsFieldsFileName := context.CLIContext.GlobalString(command.ECSFieldsFileNameFlag)
```

Comment:
What about if the user does not use the flag and instead specifies the default value?

Reply:
You could add the default value of the file name as the default value for the flag, but note that in that case you'll have to check for the existence of the file, since the default value will always be set regardless of whether or not the user has the file.

Reply:
Also be sure to check for the error case where the user does specify the flag, but the file is not found. For this reason I'd suggest not setting the default value of flag in the definition of the flag.

Reply:
This error case is handled by `readECSFields` in `convert_task_definition.go` -- where you had mentioned wrapping the error.

Reply:
Cool. I still think that the user should be allowed to have an `ecs-fields.yml` and have the CLI automatically read it even if it is not specified in the flag.